easybuild/tools/github.py

```diff
@@ -1020,6 +1048,20 @@ def merge_pr(pr):
                              pr, pr_target_account, pr_target_repo, status, reviews_data)
     pr_data['reviews'] = reviews_data
 
+    # check for required PRs mentioned in the body
+    match = re.search('[Rr]equires .*', pr_data['body'])
+    if match:
+        pr_deps = {}
+        pr_deps['framework'] = re.findall('(?:framework/pull/)(\d+)', match.group())
+        pr_deps['easyblocks'] = re.findall('(?:blocks/pull/)(\d+)', match.group())
+        pr_deps['easyconfigs'] = re.findall('(?: #|~~#)(\d+)', match.group())
+        unmerged_pr_deps = []
+        for repo in pr_deps.keys():
+            for pr_dep in pr_deps[repo]:
+                if not is_pr_merged(pr_dep, pr_target_account, 'easybuild-%s' % repo, github_user):
+                    unmerged_pr_deps.append('easybuild-%s#%s' % (repo, pr_dep))
+        pr_data['unmerged_pr_deps'] = unmerged_pr_deps
```

Comment:
@migueldiascosta Maybe it's better to flesh out this code block in a dedicated function that we can easily test in isolation (w.r.t. having multiple requirements, different repos, etc.)?

```python
def check_unmerged_pr_deps(pr_body):
    """Check for unmerged PR dependencies in specified PR body."""
    unmerged_pr_deps = []

    pr_requirements = re.search('[Rr]equires .*', pr_data['body'])
    if pr_requirements:
        ...

    return unmerged_pr_deps

def merge_pr(pr):
    ...
    pr_data['unmerged_pr_deps'] = check_unmerged_pr_deps(pr_data['body'])